core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -788,6 +788,20 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Applies a function to each partition of this RDD, while tracking the index
+   * of the original partition.
+   */
+  def foreachPartitionWithIndex(
```

Comment:
Well, this is not necessary for this particular PR, isnt it? Then no point introducing it in this PR.